### PR TITLE
Fix #102: Regression: usage of typedefs within structs.

### LINKED ISSUE #102: Regression: usage of typedefs within structs
C code:

```C
typedef struct _ProtobufCMethodDescriptor ProtobufCMethodDescriptor;

struct _ProtobufCMethodDescriptor
{
    const char *name;
    const ProtobufCMethodDescriptor *input;
    const ProtobufCMethodDescriptor *output;
};
```

Generated D code with 3e154a6d201dc8494a281ff8c9660a01194a1710 :

```D
extern (C):

struct _ProtobufCMethodDescriptor
{
    alias _ProtobufCMethodDescriptor ProtobufCMethodDescriptor;
    const(ProtobufCMethodDescriptor)* input;
    alias _ProtobufCMethodDescriptor ProtobufCMethodDescriptor;
    const(ProtobufCMethodDescriptor)* output;
}

alias _ProtobufCMethodDescriptor ProtobufCMethodDescriptor;
```

Duplicate injected alias inside a struct causes compilation error. Used to not inject such aliases before, though I haven't tracked regression commit yet.

### PR DESCRIPTION
**Pull request by ciechowoj**
As in the title.

### CODE REVIEW

**Comment by ciechowoj** [comment]
It seems like there are problems with running the tests, and more precisely downloading clang binaries.

```
Downloading clang 3.4.2
object.Exception@unit_tests/HttpClient.d(14): HTTP answered 302 instead of 200 on http://llvm.org/releases/3.4.2/clang+llvm-3.4.2-x86_64-unknown-ubuntu12.04.xz
```

**Comment by ciechowoj** [comment]
> Originally I haven't implemented redirects when removing external http client dependency as those weren't necessary. Looks like they are now. Something as simple as modifying https://github.com/jacob-carlborg/dstep/blob/master/unit_tests/HttpClient.d#L13-L14 to check 302 status and Location header in a loop should do the trick.

Good idea.

File: dub.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
             "sourcePaths": ["dstep", "clang"],
             "importPaths": ["dstep", "clang"],
             "dflags-windows-x86": ["-m32mscoff"],
-            "lflags-posix": ["-lclang", "-rpath", ".", "-rpath", "/usr/lib/llvm-3.7/lib", "-L.", "-L/usr/lib64/llvm", "-L/usr/lib/llvm-3.7/lib"],
+            "lflags-posix": ["-lclang", "-rpath", ".", "-rpath", "/usr/lib/llvm-3.8/lib", "-L.", "-L/usr/lib64/llvm", "-L/usr/lib/llvm-3.8/lib"],
```
Comment (jacob-carlborg) [inline dub.json]: Please don't modify this. I'll remove this eventually when I've finished support for static and dynamic linking of libclang.

Comment (ciechowoj) [inline dub.json]: Ok.

**Comment by jacob-carlborg** [comment]
👌

**Comment by jacob-carlborg** [comment]
Thanks.

**Pull request merged**